src/cmd/flux-job.c

```diff
@@ -1726,6 +1727,77 @@ static void finish_mpir_interface ()
     MPIR_Breakpoint ();
 }
 
+static void attach_setup_stdin (struct attach_ctx *ctx)
+{
+    flux_watcher_t *w;
+    /* flux_buffer_read_watcher_create() requires O_NONBLOCK on
+     * stdin */
+
+    if ((stdin_flags = fcntl (STDIN_FILENO, F_GETFL)) < 0)
+        log_err_exit ("fcntl F_GETFL stdin");
+    if (atexit (restore_stdin_flags) != 0)
+        log_err_exit ("atexit");
+    if (fcntl (STDIN_FILENO, F_SETFL, stdin_flags | O_NONBLOCK) < 0)
+        log_err_exit ("fcntl F_SETFL stdin");
+
+    w = flux_buffer_read_watcher_create (flux_get_reactor (ctx->h),
+                                         STDIN_FILENO,
+                                         1 << 20,
+                                         attach_stdin_cb,
+                                         FLUX_WATCHER_LINE_BUFFER,
+                                         ctx);
+    if (!w)
+        log_err_exit ("flux_buffer_read_watcher_create");
+
+    if (!(ctx->stdin_rpcs = zlist_new ()))
+        log_err_exit ("zlist_new");
+
+    ctx->stdin_w = w;
+    flux_watcher_start (ctx->stdin_w);
+}
+
+static void pty_client_wait_cb (flux_future_t *f, void *arg)
+{
+    int status;
+
+    flux_pty_client_destroy ((struct flux_pty_client *) arg);
+
+    flux_pty_client_get_wait_status (f, &status);
+    flux_future_destroy (f);
+    flux_pty_client_restore_terminal ();
+
+    /*  Hm, should we force exit here?
+     *  Need to differentiate between pty detach and normal exit.
+     */
+    log_msg ("terminal detached, exiting...");
+
+    /*  For now, just exit non-zero if pty returned nonzero status */
+    exit (status > 0 ? 1 : 0);
+}
+
+static void attach_pty (struct attach_ctx *ctx, const char *pty_service)
+{
+    int n;
+    char topic [128];
+    flux_future_t *f;
+    struct flux_pty_client *c = flux_pty_client_create ();
+    if (!c)
+        log_err_exit ("flux_pty_client_create");
+
+    n = snprintf (topic, sizeof (topic), "%s.%s", ctx->service, pty_service);
+    if (n >= sizeof (topic))
+        log_err_exit ("Failed to build pty service topic at %s.%s",
+                      ctx->service, pty_service);
+
+    log_msg ("Attaching to pty at %s", topic);
+    if (flux_pty_client_attach (c, ctx->h, ctx->leader_rank, topic) < 0)
+        log_err_exit ("flux_pty_client_attach");
+
+    if (!(f = flux_pty_client_wait (c))
```

Comment:
It is very similar to `job_wait()`, in that the future is fulfilled when the pty client completes (e.g. detaches or remote pty process exits). However, this was part of a failed experiment. I was hoping we'd be able to do something like:

 ```c
f = flux_pty_client_wait (client);
if (flux_future_get (f, NULL) < 0)
     log_err_exit ("error waiting for client to complete")
```

i.e. run a pty client in 'now' mode. Howver, this doesn't work currently. Perhaps I need to ensure all the watchers required for the client (fd and signal watchers) are started in the now reactor created by `flux_future_get` in synchronous mode.

It may not be worth it at this time, and perhaps I should just switch to a "exit_notify" mechanism as was done on the pty server side...